### PR TITLE
[stable27] fix(core): Do not invert app menu text color

### LINKED ISSUE #38759: [Bug]: Inverted app label color for light background color
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When selecting a plain color as background and choose a light one (e.g., white), the app icons in the top bar are inverted to a dark color but the labels stay white and are therefore not/poorly visible.

image

This is, because the color of the whole app menu entry is inverted for a light background color, while the `--color-primary-element-text'` is inverted as well, leading to a white text color on a light background.

### Steps to reproduce

1. In the appearance and accessibility settings select "No background" 
2. Change the color to white or a very light shade

### Expected behavior

The labels of the apps in the top bar should have a dark colour to be easily readable.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Hephi2**
Fixes #38759 
Backport of #38776

### CODE REVIEW

**Comment by mejo-** [comment]
Cypress test failures seem unrelated. The last merge to `stable27` also failed to run any Cypress e2e tests: https://github.com/nextcloud/server/actions/runs/5311915284/jobs/9615820016

**Comment by mejo-** [comment]
I have no idea why the psalm test fails. Apparently it didn't fail on former `stable27` PRs, but this PR doesn't touch anything related :thinking:

**Review by mejo- — APPROVED** [review]

File: core/src/components/AppMenu.vue [hunk 1]
```diff
@@ -146,8 +146,6 @@ $header-icon-size: 20px;
 		height: 50px;
 		position: relative;
 		display: flex;
-		opacity: .7;
```
Comment (mejo-) [inline core/src/components/AppMenu.vue:149]: This seems unrelated and a failure in the manual backport @Hephi2, no?

**Review by mejo- — CHANGES_REQUESTED** [review]

**Review by szaimen — CHANGES_REQUESTED** [review]
3rdparty should not get updated here

File: core/src/components/AppMenu.vue (hunk 1 above)
Comment (szaimen) [reply core/src/components/AppMenu.vue:149]: indeed

**Review by mejo- — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
backport looks good

**Pull request merged**